src/SingleProject/Resizetizer/src/ResizeImageInfo.cs

```diff
@@ -61,11 +63,15 @@ public static List<ResizeImageInfo> Parse(IEnumerable<ITaskItem> images)
 			foreach (var image in images)
 			{
 				var info = new ResizeImageInfo();
-
+								
 				var fileInfo = new FileInfo(image.GetMetadata("FullPath"));
+
 				if (!fileInfo.Exists)
 					throw new FileNotFoundException("Unable to find background file: " + fileInfo.FullName, fileInfo.FullName);
 
+				if (!ValidImageExtensions.Contains(fileInfo.Extension.ToUpperInvariant()))
```

Comment:
Weak validation? We could try to open the image, getting the byte array and verifying the extension information to be more precise.